newsroom/news_api/app/__init__.py

```diff
@@ -103,16 +107,122 @@ def base_exception_error(err):
 def get_app(config=None):
     app = NewsroomNewsAPI(__name__, config=config)
 
+    def convert_to_syndicate(data, token, formatter):
+        # remove token paramerters from requirments
+        if formatter and formatter == 'ATOM':
+            return NewsAPISyndicateService.generate_atom_feed(data)
+        elif formatter and formatter == 'RSS':
+            return NewsAPISyndicateService.generate_rss_feed(data)
+        elif formatter and formatter == 'JSON':
+            return jsonify(data)
+        else:
+            raise ValueError("Invalid formatter specified")
+
+    def handle_unsupported_format(data, token, formatter):
+        error_message = f"Unsupported formatter: {formatter}"
+        error_response = make_response(jsonify({'error': error_message}), 400)
+        return process_error_response(error_response)
+
+    FORMAT_HANDLERS = defaultdict(
+        lambda: {'handler': handle_unsupported_format, 'content_type': 'application/json'},
+        {
+            'ATOM': {'handler': convert_to_syndicate, 'content_type': 'application/xml'},
+            'RSS': {'handler': convert_to_syndicate, 'content_type': 'application/xml'},
+            'JSON': {'handler': convert_to_syndicate, 'content_type': 'application/json'},
+        }
+    )
+
     @app.after_request
-    def after_request(response):
+    def process_response(response):
+        if 'news/syndicate' in request.url:
+            if response.status_code >= 400:
+                return process_error_response(response)
+
+            format_param = request.args.get('formatter')
+            if format_param:
+                format_param = format_param.upper().strip()
+                try:
+                    format_handler = FORMAT_HANDLERS.get(format_param)
+                    if format_handler:
+                        token = get_auth_token()
+                        response_data = get_response_data(response)
+                        response = format_handler['handler'](response_data, token, format_param)
+                    else:
+                        error_message = f"Unsupported formatter: {format_param}"
+                        error_response = make_response(jsonify({'error': error_message}), 400)
+                        return process_error_response(error_response)
+                except ValueError as e:
+                    error_message = f"An error occurred in converting response to {format_param}: {e}"
+                    error_response = make_response(jsonify({'error': error_message}), 400)
+                    return process_error_response(error_response)
+
+        # Unconditionally add rate limit headers
+        add_rate_limit_headers(response)
+        return response
+
+    def get_auth_token() -> Optional[str]:
+        token = request.args.get('token')
```

Comment:
I wonder do we need this to look in the args for tokens?

Reply:
I can't see anywhere where `token` is used, and probably for good reason. Maybe this can be removed entirely.